api/v1alpha1/oadp_types.go

```diff
@@ -42,6 +42,7 @@ const OadpBSLProviderLabel = "openshift.io/oadp-bsl-provider"
 const DataMoverLabel = "openshift.io/volume-snapshot-mover"
 const StorageClassLabel = "openshift.io/vsm-storageclass"
 
+// +kubebuilder:validation:Enum=aws;gcp;azure;csi;vsm;openshift;kubevirt
```

Comment:
"user need to add openshift plugin" -- I don't know that we'd want to enforce this via validation. It may be that someone will want to remove it for testing purposes -- i.e. "the bug seems to be in the openshift plugin, because when I remove it, the issue goes away". Documentation should certainly tell users that under normal circumstances they should include it, but I don't think there's any reason to require it via validation.

Reply:
right, good thinking